src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -123,6 +123,29 @@ async_manifest_view_cursor::get_status() const {
       });
 }
 
+ss::future<> async_manifest_view_cursor::sync_manifest() {
+    if (std::holds_alternative<stm_manifest_t>(_current)) {
+        // Invariant: if _current points to the STM manifest the
+        //            _stm_start_offset is set
+        vassert(_stm_start_offset.has_value(), "STM start offset is not set");
+        const auto& m = std::get<stm_manifest_t>(_current).get();
+        vlog(
+          _view._ctxlog.debug,
+          "async_manifest_view_cursor sync invoked, manifest range: [{}/{}], "
+          "expected start offset: {}",
+          m.get_start_offset(),
+          m.get_last_offset(),
+          _stm_start_offset);
+        auto so = m.get_start_offset().value_or(model::offset{});
+        if (so != _stm_start_offset.value()) {
+            auto res = co_await seek(_stm_start_offset.value());
+            if (res.has_failure()) {
+                throw std::system_error(res.error());
+            }
```

Comment:
If `seek()` returns false should we throw something?

Reply:
fixed